contracts/GitHubMarket.sol

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: MPL-2.0
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+interface IMarketBehavior {
+    function authenticate(
+        address _prop,
+        string calldata _args1,
+        string calldata _args2,
+        string calldata _args3,
+        string calldata _args4,
+        string calldata _args5,
+        address market
+    )
+        external
+        returns (
+            // solium-disable-next-line indentation
+            bool
+        );
+
+    function schema() external view returns (string memory);
+
+    function getId(address _metrics) external view returns (string memory);
+
+    function getMetrics(string calldata _id) external view returns (address);
+}
+
+interface IMarket {
+    function authenticate(
+        address _prop,
+        string calldata _args1,
+        string calldata _args2,
+        string calldata _args3,
+        string calldata _args4,
+        string calldata _args5
+    )
+        external
+        returns (
+            // solium-disable-next-line indentation
+            bool
+        );
+
+    function authenticatedCallback(address _property, bytes32 _idHash)
+        external
+        returns (address);
+
+    function deauthenticate(address _metrics) external;
+
+    function schema() external view returns (string memory);
+
+    function behavior() external view returns (address);
+
+    function enabled() external view returns (bool);
+
+    function votingEndBlockNumber() external view returns (uint256);
+
+    function toEnable() external;
+}
+
+contract GitHubMarket is IMarketBehavior, Ownable {
+    address private khaos;
+    address private market;
+    bool public migratable = true;
+
+    mapping(address => string) private packages;
+    mapping(bytes32 => address) private metrics;
+    mapping(bytes32 => bool) private pendingAuthentication;
+    event Registered(address _metrics, string _package);
+    event Authenticated(AuthenticatedData _data);
+    event Query(QueryData _data);
+    struct QueryData {
+        bytes32 key;
+        string package;
+        string publicSignature;
+        address property;
+    }
+
+    struct AuthenticatedData {
+        bytes32 key;
+        string package;
+        address property;
+        int256 status;
+        string errorMessage;
+    }
```

Comment:
Khaos should also assume use cases other than Dev Protocol, so I think that information for specific use cases should be expressed as JSON-like strings.

How about express QueryData.package and QueryData.property in JSON as `'{"package": "xxx", "property": "xxx"}'`?

For example:
```solidity
struct QueryData {
    bytes32 key;
    string publicSignature;
    string additionalData; // '{"package": "xxx", "property": "xxx"}'
}
```

Similarly for AuthenticatedData:
```solidity
struct AuthenticatedData {
    bytes32 key;
    int256 status;
    string additionalData; // '{"package": "xxx", "property": "xxx", "errorMessage": "xxx"}'
}
```

Reply:
I fix that comment.

JSON is inconvenient because Solidity processes AuthenticatedData. We will need to handle those data using internal variables.

Also, since the result is not always numeric, I think the string type is appropriate.

```solidity
struct AuthenticatedData {
    bytes32 key;
    string status;
}
```